closure/goog/dom/xml.js

```diff
@@ -39,6 +40,39 @@ goog.dom.xml.MAX_ELEMENT_DEPTH = 256; // Same default as MSXML6.
 
 
 /**
+ * Check for ActiveXObject support by the browser.
+ * @return {boolean} true if browser has ActiveXObject support.
+ * @private
+ */
+goog.dom.xml.hasActiveXObjectSupport_ = function() {
+  if (!goog.userAgent.IE) {
+    // Avoid raising useless exception in case code is not compiled
+    // and browser is not MSIE.
+    return false;
+  }
+  try {
+    // Due to lot of changes in IE 9, 10 & 11 behaviour and ActiveX being
+    // totally disableable using MSIE's security level, trying to create the
+    // ActiveXOjbect is a lot more reliable than testing for the existance of
+    // window.ActiveXObject
+    new ActiveXObject('MSXML2.DOMDocument');
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+
+/**
+ * True if browser has ActiveXObject support.
+ * Possible override if this test become wrong in coming IE versions.
+ * @type {boolean}
+ */
+goog.dom.xml.ACTIVEX_SUPPORT = goog.userAgent.IE &&
```

Comment:
goog.dom.xml.ACTIVEX_SUPPORT value depends on the return value of the previously declared function...

Should I declare goog.dom.xml.ACTIVEX_SUPPORT before the function and set its value later ?